src/App.tsx

```diff
@@ -1,13 +1,20 @@
 import React from 'react';
 import './App.css';
-import BulletinList from './components/BulletinList';
-import CommentList from './components/CommentList';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+import MainPage from './pages/MainPage';
+import CommentPage from './pages/CommentPage';
 
 const App: React.FC = () => {
   return (
     <div className="App">
-      <BulletinList />
-      <CommentList postId={1} />
+      <Router>
+        <Switch>
+          <Route exact path="/">
+            <MainPage />
```

Comment:
Pass `MainPage` to `Route` as prop like you have done with `CommentPage` (which should be PostPage? :D) and then you can get rid of `Switch` and then Router works as it should